internal/handler/handler.go

```diff
@@ -0,0 +1,66 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/YerzhanAkhmetov/go-shortener/internal/errs"
+	"github.com/YerzhanAkhmetov/go-shortener/internal/usecase"
+	"github.com/gorilla/mux"
+)
+
+// Handler обрабатывает HTTP запросы
+type Handler struct {
+	usecase usecase.URLUsecase // Использование usecase для бизнес-логики URL
+	BaseURL string
+	//config *config.Config // Конфигурация приложения
+}
+
+// NewHandler создает новый экземпляр Handler
+func NewHandler(usecase usecase.URLUsecase, baseURL string) *Handler {
+	return &Handler{
+		usecase: usecase,
+		BaseURL: baseURL,
+	}
+}
+
+// CreateShortURL обрабатывает запрос на создание короткой ссылки
+func (h *Handler) CreateShortURL(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil || len(body) == 0 {
+		writeError(w, errs.NewError("Invalid request body", http.StatusBadRequest, "Bad Request"))
+		return
+	}
```

Comment:
Не стоит забывать закрывать Body  после чтения.
```suggestion
	body, err := io.ReadAll(r.Body)
	if err != nil || len(body) == 0 {
		writeError(w, errs.NewError("Invalid request body", http.StatusBadRequest, "Bad Request"))
		return
	}
	defer r.Body.Close()
```